components/zeekendpointsecurity/src/endpointsecurityconsumer.h

```diff
@@ -49,5 +49,18 @@ class EndpointSecurityConsumer final : public IEndpointSecurityConsumer {
   static Status processForkNotification(Event &event, const void *message_ptr);
 
   friend class IEndpointSecurityConsumer;
+
+  /// \brief File open event handler
+  /// \param event the generated event object
+  /// \param message_ptr a valid pointer to an EndpointSecurity es_message_t
+  /// \return A Status object
+  static Status processOpenNotification(Event &event, const void *message_ptr);
+
+  /// \brief File create event handler
+  /// \param event the generated event object
+  /// \param message_ptr a valid pointer to an EndpointSecurity es_message_t
+  /// \return A Status object
+  static Status processCreateNotification(Event &event,
+                                          const void *message_ptr);
```

Comment:
You didn't add this class and set the patterns here, but ideally we wouldn't return untyped data here, but require callers to directly pass in a `const es_message_t &message`. If you feel like it I wouldn't be opposed at all if you'd change the signatures and impls of these `static` functions in a cleanup commit.

Reply:
@bbannier could you elaborate more on this? I am confused about exactly what needs to be changed in the function signatures. Did you mean to remove parameter `const es_message_t &message` variable from functions `processCreateNotificaiton`, `processOpenNotification`, `processForkNotification` , and `processExecNotification` ?

Reply:
Sorry for being unclear. I wanted to suggest that we should not pass `const void *`, but instead `const es_message_t &` in each of these functions. This would be an orthogonal cleanup while we touch these lines.

If we did that we wouldn't need to cast from `void *` inside the implementation of these functions, and could simplify their documentation (`a valid pointer` becomes an unnecessary aspect as a reference should always need to point to valid data). If we carried that change through this would also allow us to make similar changes to the signatures and implementations of functions calling the ones here (I can see `endpointSecurityCallback` also passing untyped raw pointers).

Does that clarify that? Like I said, IMO this would be a nice to have change, nothing on the critical path of this patch.